worlds/openrct2/__init__.py

```diff
@@ -0,0 +1,526 @@
+        while (item + 7) < logic_length:
+            level = Region("OpenRCT2_Level_" + str(current_level), self.player, self.multiworld)
+            level.locations = locations_to_region(item, item + 7, level)
+            self.multiworld.regions.append(level)
+            item += 8
+            current_level += 1
+
+        level = Region("OpenRCT2_Level_" + str(current_level), self.player, self.multiworld)
+        level.locations = locations_to_region(item, (logic_length - 1), level)
+        self.multiworld.regions.append(level)
+
+        victory = Region("Victory", self.player, self.multiworld)
+        victory.locations = [OpenRCT2Location(self.player, "Victory", None, victory)]
+        self.multiworld.regions.append(victory)
+
+        r.connect(s)
+        s.connect(self.multiworld.get_region("OpenRCT2_Level_0", self.player))
+        count = 0
+        while count < current_level:
+            region = self.multiworld.get_region(f"OpenRCT2_Level_{count}", self.player)
+            region_entrance = region.connect(self.multiworld.get_region(f"OpenRCT2_Level_{count + 1}", self.player))
+            num_rides = 0
+            if count == 0:
+                pass
+            elif count == 1:  # 3 total items, we want 2 to be rides
+                num_rides = 2
+            elif count == 2:  # 7 total items, we want 4 rides and a food stall
+                num_rides = 4
+                add_rule(region_entrance, lambda state: state.has("Food Stall", self.player))
+            elif count == 3:  # 15 total items, we want 10 rides and now a drink stall
+                num_rides = 10
+                add_rule(region_entrance, lambda state: state.has("Drink Stall", self.player, 1))
+            elif count == 4:  # 23 total items, we want 15 rides and now toilets
+                num_rides = 15
+                add_rule(region_entrance, lambda state: state.has("Toilets", self.player, 1))
+            elif count == 5:  # 31 total items, we want 18 rides and some rules if applicable
+                num_rides = 18
+                if self.rules[2] == 1:  # If high construction can be disabled
+                    add_rule(region_entrance, lambda state: state.has("Allow High Construction", self.player, 1))
+                if self.rules[3] == 1:  # landscape
+                    add_rule(region_entrance, lambda state: state.has("Allow Landscape Changes", self.player, 1))
+                if self.rules[5] == 1:  # tree removal
+                    add_rule(region_entrance, lambda state: state.has("Allow Tree Removal", self.player, 1))
+                if "Cash Machine" in self.item_table:
+                    add_rule(region_entrance, lambda state: state.has("Cash Machine", self.player, 1))
+                if "First Aid" in self.item_table:
+                    add_rule(region_entrance, lambda state: state.has("First Aid", self.player, 1))
+            add_rule(region_entrance, lambda state: state.has_group("rides", self.player, num_rides))
```

Comment:
Duplicates are fine. All rides are listed as progression instead of filler, and this is to help push progression items up front to keep early sphere items from appearing super late in the game.